test/ibmq/test_ibmq_qasm_simulator.py

```diff
@@ -124,14 +123,22 @@ def _new_submit(qobj, *args, **kwargs):
                              f"qobj header={qobj.header}")
             return mock.MagicMock()
 
-        backend = copy.deepcopy(self.sim_backend)
+        backend = self.sim_backend
+
+        sim_method = backend._configuration._data.get('simulation_method', None)
+        submit_fn = backend._submit_job
+
         backend._configuration._data['simulation_method'] = 'extended_stabilizer'
         backend._submit_job = _new_submit
```

Comment:
nit - these two lines should also be inside the try block since they alter the `backend` variable, which we always want to restore.